### ISSUE TITLE
react-native-slider accessibilityState disabled

### ISSUE THREAD

**Issue by fabriziobertoglio1987** [fabriziobertoglio1987/react-native-notes#2]
Test Cases

1) Slider has `disabled` and `accessibilityState={{disabled: false}}`

```javascript
  <Slider
    disabled
    accessibilityState={{
      disabled: false,
    }}
    onValueChange={value => this.setState({value: value})}
  />
```

**Expected/Actual Result**:
The ScreenReader should announce `disabled` when the focus moves to the `Slider`.

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

<video src="https://user-images.githubusercontent.com/24992535/150911113-daaf4054-a8c5-4994-878f-050c8e699c28.mp4" width="1000" />

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
https://github.com/facebook/react-native/pull/31001/files#r578827409
>there is nothing stopping someone from setting disabled="true" accessibilityState={{disabled: false}} or vice-versa.

Test Cases

7.7 Wrong prop type disabled="not a boolean value" with flow on the master branch

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

```javascript
  <Slider disabled="not a boolean value"  />
```

<video src="https://user-images.githubusercontent.com/24992535/151465451-d6d092b4-1667-443d-a89c-89d01e0f1c79.mp4" width="1000" />

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
https://github.com/facebook/react-native/pull/31001/files#r578827409
>there is nothing stopping someone from setting disabled="true" accessibilityState={{disabled: false}} or vice-versa.

Test Cases

7.6 Wrong prop type disabled="not a boolean value" with typescript

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

```javascript
  <Slider disabled="not a boolean value"  />
```

https://user-images.githubusercontent.com/24992535/151469374-e36debf0-cc46-44b0-bab6-ecc4fca90b75.mp4


<video src="" width="1000" />

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
https://github.com/facebook/react-native/pull/31001/files#r578827409
>there is nothing stopping someone from setting disabled="true" accessibilityState={{disabled: false}} or vice-versa.

Test Cases

7.5 Slider has disabled={false} and accessibilityState={{disabled:true}} on master

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

```javascript
  <Slider
    disabled={false}
    accessibilityState={{disabled: true}}
    onValueChange={value => this.setState({value: value})}
  />
```

<video src="" width="1000" />

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
Test Cases

7.4 Slider has accessibilityState={{disabled:false}} on master

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

```javascript
  <Slider
    accessibilityState={{disabled: false}}
    onValueChange={value => this.setState({value: value})}
  />
```

<video src="" width="1000" />

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
Test Cases

7.3. Slider has `accessibilityState={{disabled: true}}` on master

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

```javascript
  <Slider
    accessibilityState={{disabled: true}}
    onValueChange={value => this.setState({value: value})}
  />
```

<video src="" width="1000" />

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
Test Cases

7.1 Slider has `disabled` and `accessibilityState={{disabled: false}}` on main

**<details><summary>CLICK TO OPEN TESTS RESULTS</summary>**
<p>

```javascript
  <Slider
    disabled
    accessibilityState={{
      disabled: false,
    }}
    onValueChange={value => this.setState({value: value})}
  />
```

<video src="https://user-images.githubusercontent.com/24992535/151501304-556abb86-1fe8-4025-886a-f49ad3994ed8.mp4" width="1000" />

</p>
</details>

**Comment by fabOnReact** [comment]
### TO-DO
- [ ] Test iOS
- [ ] Add test cases to react-native-slider pull request

**Comment by fabOnReact** [comment]
PR https://github.com/callstack/react-native-slider/pull/354
